### PR TITLE
feat: bar graph entity

### PR DESCRIPTION
**Pull request by gitanjli525**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
bar graph entity added
image

## Motivation and Context
new graph implementation for payment analytics

## How did you test it?
core changes, no UI changes in this PR

## Where to test it?

- [x] INTEG
- [x] SANDBOX
- [ ] PROD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/NewAnalytics/Graphs/BarGraph/BarGraph.res [hunk 1]
```diff
@@ -0,0 +1,11 @@
+external barGraphOptionsToJson: BarGraphTypes.barGraphOptions => JSON.t = "%identity"
```
Comment (PritishBudhiraja) [inline src/screens/NewAnalytics/Graphs/BarGraph/BarGraph.res:1]: Move to identity file

File: src/screens/NewAnalytics/Graphs/BarGraph/BarGraphTypes.res [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+type tickmarkPlacement = string
+type endOnTick = bool
+type startOnTick = bool
+type tickInterval = int
+type tickWidth = int
+type min = int
+type max = int
+type showInLegend = bool
+type name = string
+
```
Comment (PritishBudhiraja) [inline src/screens/NewAnalytics/Graphs/BarGraph/BarGraphTypes.res:20]: why separate types?
Any specific usecase? 
I'm not able to find anywhere these used and if the type can be directly placed

File: src/screens/NewAnalytics/Graphs/BarGraph/BarGraph.res (hunk 1 above)
Comment (gitanjli525) [reply src/screens/NewAnalytics/Graphs/BarGraph/BarGraph.res:1]: this is not used anywhere else other than here, so we can keep it here itself

Comment (PritishBudhiraja) [reply src/screens/NewAnalytics/Graphs/BarGraph/BarGraph.res:1]: The identity file is made for that purpose to keep the identity at some place it's not about reusing

Comment (PritishBudhiraja) [reply src/screens/NewAnalytics/Graphs/BarGraph/BarGraph.res:1]: And if we are using Identy function only then why can't we use this only

```js
Identity.genericTypeToJson
```

As this also converting to json if we are by passing types it shouldn't even matter what is it before so we can use this only

Comment (JeevaRamu0104) [reply src/screens/NewAnalytics/Graphs/BarGraph/BarGraph.res:1]: we don't want to expose the BarGraphTypes to Identity file and this is specific to Bar graph so we decided to kept it over here itself.Later we will see whether to move or not

File: src/screens/NewAnalytics/Graphs/BarGraph/BarGraphTypes.res (hunk 2 above)
Comment (JeevaRamu0104) [reply src/screens/NewAnalytics/Graphs/BarGraph/BarGraphTypes.res:20]: we are having some types common to LineGraph and BarGraph which we will move it to common place

**Review by JeevaRamu0104 — DISMISSED** [review]

File: src/screens/NewAnalytics/Graphs/BarGraph/BarGraph.res (hunk 1 above)
Comment (PritishBudhiraja) [reply src/screens/NewAnalytics/Graphs/BarGraph/BarGraph.res:1]: Hey @JeevaRamu0104 if we are by passing types it's okay to use this. Why we are adding this then. It won't add any safety
```
Identity.genericTypeToJson
```

**Pull request merged**